packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -318,12 +369,34 @@ export class OdspDocumentService implements IDocumentService {
                 this.epochTracker,
                 requestSocketToken,
                 options,
+                refreshJoinSessionPolicy,
                 this.hostPolicy.sessionOptions?.unauthenticatedUserDisplayName,
             );
 
-        // Note: The sessionCache is configured with a sliding expiry of 1 hour,
-        // so if we've fetched the join session within the last hour we won't run executeFetch again now.
-        return this.cache.sessionJoinCache.addOrGet(this.joinSessionKey, executeFetch);
+        let response: ISocketStorageDiscovery;
+        const currentTime = Date.now();
+        // If the join session refresh time is in future, that means we can reuse the cached response. Otherwise
+        // we need to clear the cache and execute the network call, so as to refresh the session on the relay
+        // ordering service.
+        if (this.joinSessionRefreshTime === undefined || this.joinSessionRefreshTime <= currentTime) {
+            this.cache.sessionJoinCache.remove(this.joinSessionKey);
+            response = await this.cache.sessionJoinCache.addOrGet(this.joinSessionKey, executeFetch);
+            if (!refreshJoinSessionPolicy) {
+                // This is for the old flow. If the refresh policy is not set, then keep the response for 1 hour.
+                this.joinSessionRefreshTime = currentTime + 3600000;
```

Comment:
I'd suggest move this statement up (that's default value), as you want to set it in case refreshJoinSessionPolicy === true && refreshSessionDurationSeconds  === undefined below